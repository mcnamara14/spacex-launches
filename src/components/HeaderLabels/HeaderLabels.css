.HeaderLabels {
  height: 10px;
  display: flex;
  padding: 20px;
  margin-top: 1px;
  align-items: center;
  justify-content: space-between;
  background: rgb(159,184,206); 
  background: -moz-linear-gradient(top, rgba(159,184,206,1) 0%, rgba(222,231,238,1) 36%, rgba(255,255,255,1) 68%, rgba(255,255,255,1) 100%); 
  background: -webkit-linear-gradient(top, rgba(159,184,206,1) 0%,rgba(222,231,238,1) 36%,rgba(255,255,255,1) 68%,rgba(255,255,255,1) 100%); 
  background: linear-gradient(to bottom, rgba(159,184,206,1) 0%,rgba(222,231,238,1) 36%,rgba(255,255,255,1) 68%,rgba(255,255,255,1) 100%); 
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#9fb8ce', endColorstr='#ffffff',GradientType=0 );
}

h2 {
  font-size: 16px;
  color: #043d67;
  margin-right: 30px;
  margin-top: 31px;
  width: auto;
  padding-left: 5px;
  height: 28px;
  padding-right: 5px;
  margin-bottom: 18px;
}

h2.date:hover, .refreshIcon {
  cursor: pointer;
}

h2.date.desc {
  margin-bottom: 18px;
  border-top: 4px solid #27d9cf;
  padding-top: 14px;
  margin-top: 13px;
}

h2.date.asc {
  margin-bottom: 14px;
  border-bottom: 4px solid #27d9cf;
}

.badge {
  width: 50px;
}

.name {
  width: 94px;
}

.type {
  width: 87px;
}

.date {
  width: 86px;
}

.details {
  width: 46px;
  margin-right: 218px !important;
}

.id {
  width: 15px;
}

.article {
  width: 44px;
  margin-right: 0 !important;
}