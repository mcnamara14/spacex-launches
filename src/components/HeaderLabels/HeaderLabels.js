import React, { Component } from 'react';
import { connect } from 'react-redux';
import './HeaderLabels.css';
import { toggleDateSort } from '../../actions/toggleDateSort';

class HeaderLabels extends Component {
  toggleSort = () => {
    this.props.toggleDateSort();
  }

  render() {
    return (
      <section className="HeaderLabels" >
          <h2 className="badge">Badge</h2>
          <h2 className="name">Rocket Name</h2>
          <h2 className="type">Rocket Type</h2>
          <h2 className="date" onClick={() => this.toggleSort()}>Launch Date</h2>
          <h2 className="details">Details</h2>
          <h2 className="id">ID</h2>
          <h2 className="article">Article</h2>
      </section>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  toggleDateSort: () => dispatch(toggleDateSort())
})

export default connect(null, mapDispatchToProps)(HeaderLabels);
