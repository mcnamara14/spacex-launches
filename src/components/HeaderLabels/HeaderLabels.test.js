import React from 'react';
import { HeaderLabels } from './HeaderLabels';
import { shallow } from 'enzyme';

describe('HeaderLabels', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<HeaderLabels toggleDateSort={jest.fn()}/>);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleClick', () => {
    it('should set state to true when it is false', () => {
      wrapper.instance().handleClick()

      expect(wrapper.instance().state.sort).toEqual(true)
    })

    it('should set state to false when it is true', () => {
      wrapper.setState({sort: true})
      wrapper.instance().handleClick()

      expect(wrapper.instance().state.sort).toEqual(false)
    })
  })
});