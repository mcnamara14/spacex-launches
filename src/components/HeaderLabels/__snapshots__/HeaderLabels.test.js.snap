// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderLabels should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeaderLabels
    toggleDateSort={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="badge"
>
          Badge
</h2>,
        <h2
          className="name"
>
          Rocket Name
</h2>,
        <h2
          className="type"
>
          Rocket Type
</h2>,
        <h2
          className="date desc"
          onClick={[Function]}
>
          Launch Date
</h2>,
        <h2
          className="details"
>
          Details
</h2>,
        <h2
          className="id"
>
          ID
</h2>,
        <h2
          className="article"
>
          Article
</h2>,
      ],
      "className": "HeaderLabels",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Badge",
          "className": "badge",
        },
        "ref": null,
        "rendered": "Badge",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Rocket Name",
          "className": "name",
        },
        "ref": null,
        "rendered": "Rocket Name",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Rocket Type",
          "className": "type",
        },
        "ref": null,
        "rendered": "Rocket Type",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Launch Date",
          "className": "date desc",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Launch Date",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Details",
          "className": "details",
        },
        "ref": null,
        "rendered": "Details",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "ID",
          "className": "id",
        },
        "ref": null,
        "rendered": "ID",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Article",
          "className": "article",
        },
        "ref": null,
        "rendered": "Article",
        "type": "h2",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="badge"
>
            Badge
</h2>,
          <h2
            className="name"
>
            Rocket Name
</h2>,
          <h2
            className="type"
>
            Rocket Type
</h2>,
          <h2
            className="date desc"
            onClick={[Function]}
>
            Launch Date
</h2>,
          <h2
            className="details"
>
            Details
</h2>,
          <h2
            className="id"
>
            ID
</h2>,
          <h2
            className="article"
>
            Article
</h2>,
        ],
        "className": "HeaderLabels",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Badge",
            "className": "badge",
          },
          "ref": null,
          "rendered": "Badge",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Rocket Name",
            "className": "name",
          },
          "ref": null,
          "rendered": "Rocket Name",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Rocket Type",
            "className": "type",
          },
          "ref": null,
          "rendered": "Rocket Type",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Launch Date",
            "className": "date desc",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Launch Date",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Details",
            "className": "details",
          },
          "ref": null,
          "rendered": "Details",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ID",
            "className": "id",
          },
          "ref": null,
          "rendered": "ID",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Article",
            "className": "article",
          },
          "ref": null,
          "rendered": "Article",
          "type": "h2",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
